File: backend/src/main/java/com/fooddelivery/service/DeliveryService.java
package com.fooddelivery.service;

import com.fooddelivery.dto.response.OrderResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

/**
 * Delivery-agent-specific operations.
 */
public interface DeliveryService {

    /** Returns orders that are ready for pickup and have no agent assigned yet. */
    List<OrderResponse> getAvailableOrders();

    /** Returns all orders assigned to the current delivery agent. */
    Page<OrderResponse> getMyDeliveries(Pageable pageable);

    /** Agent self-assigns to an unassigned order. */
    OrderResponse acceptDelivery(Long orderId);

    /** Returns the full order for an assigned delivery agent to view. */
    OrderResponse getDeliveryStatus(Long orderId);

    /** Assigns a delivery agent to an order (ADMIN only). */
    OrderResponse assignDelivery(Long orderId, Long deliveryAgentId);

    /** Marks an order as DELIVERED. */
    OrderResponse markDelivered(Long orderId);
}

File: frontend/src/services/deliveryService.js
/**
 * Delivery Service
 * Matches DeliveryController endpoints at /api/delivery
 */
import api from './api'

const deliveryService = {
  /** Orders ready for pickup with no agent assigned yet */
  getAvailableOrders: () => api.get('/delivery/available'),

  /** My assigned deliveries (paginated) */
  getMyDeliveries: (params = { page: 0, size: 20 }) =>
    api.get('/delivery/my-deliveries', { params }),

  /** Self-assign to an available order */
  acceptDelivery: (orderId) => api.patch(`/delivery/${orderId}/accept`),

  /** Mark an assigned order as delivered */
  markDelivered: (orderId) => api.patch(`/delivery/${orderId}/delivered`),
}

export default deliveryService
